### PR TITLE
Fix namespace deletion.

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

**Comment by leahnp** [comment]
Are all k8s "namespaces" "service namespaces"?

**Comment by davidewatson** [comment]
Note that the change to the `when` clause is a change in behavior. We used to unconditionally attempt to delete namespaces and then ignored any errors. Now we only delete namespaces on down. This might leak namespaces, but I think that is still possible if a user removes a chart from the kraken config and there are no other charts with that namespace.

File: ansible/roles/kraken.services/tasks/kill-services.yaml [hunk 1]
```diff
@@ -90,9 +90,9 @@
 
 - name: Delete all service namespaces
   command: >
-    {{ kubectl }} --kubeconfig={{ kubeconfig }} delete namespace {{ item }}
+    {{ kubectl }} --kubeconfig={{ kubeconfig }} delete namespace {{ item.namespace }}
   with_items: "{{ cluster_namespaces }}"
-  when: cluster_namespaces is defined
+  when: kraken_action == 'down'
```
Comment (joejulian) [inline ansible/roles/kraken.services/tasks/kill-services.yaml]: Why bother? If the cluster is being brought down, there will be no etcd data left anyway.

Comment (davidewatson) [reply ansible/roles/kraken.services/tasks/kill-services.yaml]: "The only way to be sure."

Comment (davidewatson) [reply ansible/roles/kraken.services/tasks/kill-services.yaml]: Ack.

**Pull request merged**